components/servo/lib.rs

```diff
@@ -527,14 +540,24 @@ impl Servo {
         );
 
         Servo {
+            delegate: Rc::new(DefaultServoDelegate),
             compositor: Rc::new(RefCell::new(compositor)),
             constellation_proxy: ConstellationProxy::new(constellation_chan),
             embedder_receiver,
-            messages_for_embedder: Vec::new(),
+            messages_for_embedder: Default::default(),
+            webviews: Default::default(),
             _js_engine_setup: js_engine_setup,
         }
     }
 
+    pub fn delegate(&self) -> Rc<dyn ServoDelegate> {
+        self.delegate.clone()
+    }
+
+    pub fn set_delegate(&mut self, delegate: Rc<dyn ServoDelegate>) {
+        self.delegate = delegate;
+    }
```

Comment:
Do we need to set the delegate dynamically? I feel like it can just be a parameter for `new` method.

Reply:
Yes, this is a pretty central part of the API. The idea is that the `WebView` should work even with no delegate set. The idea being that the only thing you need to do to get a working application is to create a `WebView` and add it to your interface.

Reply:
This is essentially the same pattern that WebKit uses FWIW: https://stackoverflow.com/questions/57144195/uidelegate-method-create-webview-with-configuration-not-called